action/main.go

```diff
@@ -124,16 +124,33 @@ submodule_out:
 	}
 
 	// Lets build stuff
-	_, err = recipes.Build(
+	results, err := recipes.Build(
 		ctx,
 		CLI.Build.Target,
 		CLI.Build.Recursive,
 		CLI.Build.Interactive,
 		myConfig,
 		recipes.Execute,
 	)
+
+	// Print overview
+	summary := "Build summary:"
+	for _, item := range results {
+		result := ""
+		if item.BuildResult == nil {
+			result = "Success"
+		} else if errors.Is(item.BuildResult, recipes.ErrBuildSkipped) {
+			result = "Skipped"
+		} else {
+			result = "Fail"
+		}
+		summary = fmt.Sprintf("%s\n%s: %s", summary, item.Name, result)
+	}
+	slog.Info(summary)
+
 	if err == nil {
```

Comment:
This conditional here at the end is not really the 'Go way' to do it and I also don't know why I haven't noticed this before (probably because the check is so small): Normally we just test if `err != nil` and only branch out on error conditions and simply return nil at the functions end. Given that there is no extra logic other than this in the conditional block it makes no difference but for the future I'd also recommend having the 'happy path' in indentation level 0 and every error path inside a conditional.

Reply:
Yeah, than if statement is not the proudest piece of code, when looking at it now. The `return nil` is completely unnecessary.

Reply:
Oh, now I see, it is a bit of leftover from previous commits. [This was the original](https://github.com/9elements/firmware-action/commit/ee19f2945403e99e093600fe3d5411d93abffd5a).

Reply:
I just removed the `return nil`. I hope that now it is more acceptable.